src/telemetry.c

```diff
@@ -977,17 +977,22 @@ int tm_set_payload(struct telem_ref *t_ref, char *payload)
         size_t payload_len;
         int ret = 0;
 
-        payload_len = strlen((char *)payload);
-
-        if (payload_len > MAX_PAYLOAD_LENGTH) {
-                return -EINVAL;
+        if (payload == NULL) {
+                telem_log(LOG_WARNING, "payload pointer is NULL\n");
+                return -ENOMEM;
         }
 
+        payload_len = strnlen(payload, MAX_PAYLOAD_LENGTH);
```

Comment:
So previously it would be an error and incomplete data wouldn't be sent. Is it better to send partial data where whatever was at the end of the string is discarded?

Reply:
Yes, at the very least this is a signal that we have something not quite right.

Reply:
Maybe in the future consider adding a token for we clipped data so we can know if we didn't actually get all of the payload but good for now.